bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/package.scala

```diff
@@ -1,27 +1,24 @@
 package edu.gemini.ags
 
-import edu.gemini.spModel.core.{MagnitudeBand, Magnitude}
-import edu.gemini.spModel.core.Target.SiderealTarget
-import edu.gemini.spModel.gemini.obscomp.SPSiteQuality.Conditions
+import edu.gemini.catalog.api.{SingleBandExtractor, FirstBandExtractor, MagnitudeExtractor}
+import edu.gemini.spModel.core.MagnitudeBand
+
+import scalaz._
+import Scalaz._
 
 package object api {
-  // Function that can extract a magnitude out of a target
-  type MagnitudeExtractor = SiderealTarget => Option[Magnitude]
 
-  val RLikeBands = List(MagnitudeBand._r, MagnitudeBand.R, MagnitudeBand.UC)
+  val RLikeBands = NonEmptyList(MagnitudeBand._r, MagnitudeBand.R, MagnitudeBand.UC)
 
-  /**
-   * For a given target set of probe bands build a MagnitudeExtractor that returns the first magnitude on the target
-   */
-  def magnitudeExtractor(probeBands: List[MagnitudeBand]): MagnitudeExtractor = (st: SiderealTarget) => probeBands.flatMap(st.magnitudeIn).headOption // Picks the first available magnitude on the target
+  def agsBandExtractor(band: MagnitudeBand): MagnitudeExtractor = if (band === MagnitudeBand.R) FirstBandExtractor(RLikeBands) else SingleBandExtractor(band)
```

Comment:
That is true, it could be a single class doing both, R-family and H.
At some point it has to be encoded that R expands to r, R, UC and this is the place. I'm open to better suggestions on how to encode that fact but sooner or later you need to express that R expands to the other bands